libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/DBSchemaAccessorFactory.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.tools.dbbrowser.schema;
+
+import java.util.Map;
+
+import org.apache.commons.lang3.Validate;
+import org.springframework.jdbc.core.JdbcOperations;
+
+import com.oceanbase.tools.dbbrowser.AbstractDBBrowserFactory;
+import com.oceanbase.tools.dbbrowser.schema.doris.DorisSchemaAccessor;
+import com.oceanbase.tools.dbbrowser.schema.mysql.MySQLNoLessThan5600SchemaAccessor;
+import com.oceanbase.tools.dbbrowser.schema.mysql.MySQLNoLessThan5700SchemaAccessor;
+import com.oceanbase.tools.dbbrowser.schema.mysql.OBMySQLBetween220And225XSchemaAccessor;
+import com.oceanbase.tools.dbbrowser.schema.mysql.OBMySQLBetween2260And2276SchemaAccessor;
+import com.oceanbase.tools.dbbrowser.schema.mysql.OBMySQLBetween2277And3XSchemaAccessor;
+import com.oceanbase.tools.dbbrowser.schema.mysql.OBMySQLNoGreaterThan1479SchemaAccessor;
+import com.oceanbase.tools.dbbrowser.schema.mysql.OBMySQLSchemaAccessor;
+import com.oceanbase.tools.dbbrowser.schema.mysql.ODPOBMySQLSchemaAccessor;
+import com.oceanbase.tools.dbbrowser.schema.oracle.OBOracleBetween4000And4100SchemaAccessor;
+import com.oceanbase.tools.dbbrowser.schema.oracle.OBOracleLessThan2270SchemaAccessor;
+import com.oceanbase.tools.dbbrowser.schema.oracle.OBOracleLessThan400SchemaAccessor;
+import com.oceanbase.tools.dbbrowser.schema.oracle.OBOracleSchemaAccessor;
+import com.oceanbase.tools.dbbrowser.schema.oracle.OracleSchemaAccessor;
+import com.oceanbase.tools.dbbrowser.util.ALLDataDictTableNames;
+import com.oceanbase.tools.dbbrowser.util.VersionUtils;
+
+import lombok.Setter;
+import lombok.experimental.Accessors;
+
+@Setter
+@Accessors(chain = true)
+public class DBSchemaAccessorFactory extends AbstractDBBrowserFactory<DBSchemaAccessor> {
+
+    public static final String TENANT_NAME_KEY = "tenantName";
+    public static final String SYS_OPERATIONS_KEY = "sysOperations";
+
+    private String dbVersion;
+    private JdbcOperations jdbcOperations;
```

Comment:
use datasource instead of jdbcOperation

Reply:
I have tried it, but there is a problem: we can not get datasource directly from `ConnectionSession`.
there is a util helping us get schema accessor based on `ConnectionSession`:
```java
// DBSchemaAccessors#create
public static DBSchemaAccessor create(ConnectionSession connectionSession, String dataSourceName) {
        ...
        return create(syncJdbcExecutor, sysSyncJdbcExecutor, connectType, dbVersion, tenantName);
    }

    public static DBSchemaAccessor create(@NonNull JdbcOperations syncJdbcExecutor, JdbcOperations sysJdbcExecutor,
            @NonNull ConnectType connectType, @NonNull String dbVersion, String tenantName) {
        Map<String, Object> properties = new HashMap<>();
        properties.put(DBSchemaAccessorFactory.TENANT_NAME_KEY, tenantName);
        properties.put(DBSchemaAccessorFactory.SYS_OPERATIONS_KEY, sysJdbcExecutor);
        return DBBrowser.schemaAccessor()
                .setProperties(properties)
                .setDbVersion(dbVersion)
                .setJdbcOperations(syncJdbcExecutor)
                .setType(connectType.getDialectType().getDBBrowserDialectTypeName())
                .create();
    }
```

we need to change `connectionSession` to get datasource  for this situation bug I think we can not benefit a log from it~